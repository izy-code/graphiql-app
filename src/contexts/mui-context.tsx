'use client';

import { createTheme, ThemeProvider } from '@mui/material/styles';
import type { ReactNode } from 'react';

const theme = createTheme({
  typography: {
    allVariants: {
      fontFamily: ['Inter', 'system-ui', 'Avenir', 'Helvetica', 'Arial', 'sans-serif'].join(','),
      textTransform: 'none',
      fontSize: 18,
    },
  },
});

export function MuiProvider({ children }: { children: ReactNode }): ReactNode {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}
