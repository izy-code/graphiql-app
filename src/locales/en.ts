import { MIN_PASSWORD_LENGTH } from '@/common/constants';

export default {
  'header.links.sign-in': 'Sign in',
  'header.links.sign-up': 'Sign up',
  'header.button.sign-out': 'Sign out',

  'main.welcome': 'Welcome, please sign in.',
  'main.welcome.user': 'Welcome, {name}!',
  'main.title': 'REST/GraphQL client',
  'main.links.rest': 'REST client',
  'main.links.graphql': 'GraphQL client',
  'main.links.history': 'History',
  'main.links.sign-in': 'Sign in',
  'main.links.sign-up': 'Sign up',
  'main.project': 'Project information',
  'main.project-info':
    'Developed as the final assignment for a React course, this project showcases the coordinated effort of three dedicated participants.\nCompleted within a span of four weeks, the application is a result of the team’s hard work and collaboration.\nThe application supports both RESTful and GraphQL APIs, featuring Firebase authentication, request history tracking, and language switching.\nUsers can configure and send requests, view responses.',
  'main.team.name-1': 'Ivan Zhirebny',
  'main.team.role-1': 'Frontend-developer, Team leader',
  'main.team.contributions-1': 'Dev environment setup, Authentication pages, Firebase, GraphQL',
  'main.team.bio-1':
    'Graduated from the Bauman Moscow State Technical University, completed the Frontend Developer program at the HTML Academy',
  'main.team.name-2': 'Vadim Kolymbet',
  'main.team.role-2': 'Frontend-developer',
  'main.team.contributions-2': 'Routing, Internationalization, Testing',
  'main.team.bio-2':
    'Graduated from the Saratov State University, studied at the Faculty of Computer Science and Information Technologies and in the Department of Computer Security and Cryptography Theory. Graduated from layout courses at Frontendblok. Worked as a T-SQL developer',
  'main.team.name-3': 'Aleksandra Bodnar',
  'main.team.role-3': 'Frontend-developer, Scrum-master',
  'main.team.contributions-3': 'Design, Jira, MUI, REST, History page, Welcome page',
  'main.team.bio-3':
    'Graduated from the Immanuel Kant Baltic Federal University, studied at the Faculty of Business Informatics. Graduated from 3WC courses with a degree in Frontend-development. Built PHP-driven features for 1C CRM',
  'main.rs-1':
    'RSSchool is an online educational community focused on web development and programming, offering well-structured courses for both beginners and experienced developers.',
  'main.rs-2':
    'Renowned for its high-quality education, RSSchool provides a comprehensive curriculum covering everything from basic HTML, CSS, and JavaScript to advanced technologies like React and Node.js. The school emphasizes project-based learning, enabling students to build real-world projects and develop impressive portfolios.',
  'main.rs-3':
    'RSSchool thrives on a vibrant community of volunteers, including seasoned developers and alumni, who mentor and support new learners.',

  'teamMember.role': 'Role',
  'teamMember.contributions': 'Contributions',
  'teamMember.bio': 'Short bio',
  'teamMember.techStack': 'Tech stack: ',

  'sign-in.loader': 'Signing in...',
  'sign-in.title': 'Sign in',
  'sign-in.email': 'Email',
  'sign-in.password': 'Password',
  'sign-in.submit': 'Sign in',

  'sign-up.loader': 'Signing up...',
  'sign-up.title': 'Sign up',
  'sign-up.name': 'Name',
  'sign-up.email': 'Email',
  'sign-up.password.title': 'Password',
  'sign-up.password': 'Enter password',
  'sign-up.password.confirm': 'Confirm password',
  'sign-up.submit': 'Sign up',

  'rest.request.completed': 'The request has been completed, check the response field',
  'rest.title': 'REST client',
  'rest.endpoint': 'Endpoint URL',
  'rest.params': 'Params',
  'rest.headers': 'Headers',
  'rest.variables': 'Variables',
  'rest.body.title': 'Body: ',
  'rest.body': 'Body',

  'restApi.errors.fetch': 'Failed to fetch the response',
  'restApi.errors.endpoint': 'No endpoint provided',
  'restApi.errors.body': 'Response body contains errors',
  'restApi.errors.unknown': 'Unknown error occurred while making the request',
  'restApi.errors.status': 'Fetch error',

  method: 'Method',

  'clientTable.hide': 'Hide',
  'clientTable.show': 'Show',
  'clientTable.key': 'Key',
  'clientTable.value': 'Value',
  'clientTable.row': '+ row',
  'clientTable.delete': 'Delete',

  'requestButton.send': 'Send Request',

  'response.title': 'Response',
  'response.status': 'Status:',
  'response.body': 'Body:',

  'graphql.errors.parse': 'Can’t parse request body in URL',
  'graphql.schema.completed': 'The request has been completed, you can open the schema',
  'graphql.request.completed': 'The request has been completed, check the response field',
  'graphql.title': 'GraphQL client',
  'graphql.endpoint': 'Endpoint URL',
  'graphql.schema': 'Schema URL',
  'graphql.request': 'Request',
  'graphql.schema.request': 'Schema request',
  'graphql.schema.hide': 'Hide schema',
  'graphql.schema.show': 'Show schema',
  'graphql.params': 'Params',
  'graphql.query.title': 'Query: ',
  'graphql.query': 'Query',
  'graphql.headers': 'Headers',
  'graphql.variables.title': 'Variables: ',
  'graphql.variables': 'Variables',

  'schema.title': 'Schema docs:',

  'graphQlApi.response-errors.endpoint': 'No endpoint or query provided',
  'graphQlApi.response-errors.variables': 'Variables field is not valid JSON',
  'graphQlApi.response-errors.data': 'Unknown response structure: no "data" field. Check endpoint and request params',
  'graphQlApi.response-errors.unknown': 'Unknown error occurred while making the request',
  'graphQlApi.response-errors.status': 'Fetch error',

  'graphQlApi.schema-errors.endpoint': 'No schema endpoint provided',
  'graphQlApi.schema-errors.body': 'Schema docs response body contains errors',
  'graphQlApi.schema-errors.data':
    'Unknown response structure: no "data" or "__schema" fields. Check endpoint and headers',
  'graphQlApi.schema-errors.fetch-status': 'Schema docs fetch failed with status code: {status}',
  'graphQlApi.schema-errors.unknown': 'Unknown error occurred while making the schema request',

  'history.title': 'History',
  'history.empty.title': 'You haven’t executed any requests',
  'history.empty': 'It’s empty here. Try: ',
  'history.links.rest': 'REST client',
  'history.links.graphql': 'GraphQL client',
  'history.subtitle': 'Requests:',

  'error.title': 'Oops!',
  'error.text': 'Sorry, an unexpected error has occurred.',
  'error.desc': 'Error message:',
  'error.recommendation': 'Please try to refresh the page.',
  'error.refresh': 'Refresh the page',

  '404.message': 'The page you requested was not found.',
  '404.back': 'Previous page',
  '404.main': 'Main page',

  'auth.loading': 'Loading Firebase...',
  'auth.expired': 'Your token has expired, please sign in again',
  'auth.redirect': 'Redirecting...',

  'firebase.sign-out.success': 'You have been signed out. History of requests are cleared!',
  'firebase.sign-in.success': 'Successfully signed in with {email}!',
  'firebase.sign-up.success': 'Successfully signed up {name}!',
  'firebase.errors.USER_DISABLED': 'User disabled',
  'firebase.errors.INVALID_LOGIN_CREDENTIALS': 'Invalid credentials',
  'firebase.errors.EMAIL_EXISTS': 'Email already in use',
  'firebase.errors.TOO_MANY_ATTEMPTS_TRY_LATER': 'Too many authentication requests',
  'firebase.errors.NETWORK_REQUEST_FAILED': 'Network request failed',

  'validation.email.required': 'Email is required',
  'validation.email.format': 'Email must have valid format',
  'validation.password.required': 'Password is required',
  'validation.password.number': 'Password must contain a number',
  'validation.password.uppercase': 'Password must contain an uppercase letter',
  'validation.password.lowercase': 'Password must contain a lowercase letter',
  'validation.password.special': 'Password must contain one of: -+/%*:#@\\$!?|^&',
  'validation.password.min': `Password must be at least ${MIN_PASSWORD_LENGTH} characters`,
  'validation.name.required': 'Name is required',
  'validation.name.capital': 'Name must start with a capital letter',
  'validation.confirm.required': 'Please confirm your password',
  'validation.confirm.match': 'Passwords do not match',

  'customTextarea.prettifyError': 'Failed to prettify code. Check {language} syntax',
  'customTextarea.show': 'Show',
  'customTextarea.hide': 'Hide',
  'customTextarea.plainText': 'Plain text',
  'customTextarea.format': 'Prettify',
} as const;
