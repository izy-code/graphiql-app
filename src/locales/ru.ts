import { MIN_PASSWORD_LENGTH } from '@/common/constants';

export default {
  'header.links.sign-in': 'Войти',
  'header.links.sign-up': 'Зарегистрироваться',
  'header.button.sign-out': 'Выйти',

  'main.welcome': 'Добро пожаловать, пожалуйста, войдите в систему.',
  'main.signed-in': 'С возвращением, {name}!',
  'main.title': 'REST/GraphQL клиент',
  'main.links.rest': 'REST клиент',
  'main.links.graphql': 'GraphQL клиент',
  'main.links.history': 'История',
  'main.links.sign-in': 'Войти',
  'main.links.sign-up': 'Зарегистрироваться',
  'main.project': 'Информация о проекте',
  'main.project-info':
    'Разработанный в качестве итогового задания на курсе React, этот проект демонстрирует скоординированные усилия трёх преданных участников.\nЗавершённое в течение четырёх недель, это приложение является результатом усердной работы и сотрудничества команды.\nПриложение поддерживает как RESTful, так и GraphQL API, включает аутентификацию через Firebase, отслеживание истории запросов и переключение языков.\nПользователи могут настраивать и отправлять запросы, просматривать ответы.',
  'main.team.name-1': 'Иван Жиребный',
  'main.team.role-1': 'Frontend-разработчик, Лидер команды',
  'main.team.contributions-1': 'Настройка среды разработки, Страницы аутентификации, Firebase, GraphQL',
  'main.team.bio-1':
    'Окончил Московский государственный технический университет имени Баумана, завершил профессию «Фронтенд-разработчик» в HTML Academy.',
  'main.team.name-2': 'Вадим Колымбет',
  'main.team.role-2': 'Frontend-разработчик',
  'main.team.contributions-2': 'Маршрутизация, Интернационализация, Тестирование',
  'main.team.bio-2':
    'Окончил Саратовский государственный университет, обучался на факультете Компьютерных Наук и Информационных Технологий, а также на кафедре Теоретических основ Компьютерной Безопасности и Криптографии. Завершил курсы по верстке от Frontendblok. Работал T-SQL разработчиком.',
  'main.team.name-3': 'Александра Боднар',
  'main.team.role-3': 'Frontend-разработчик, Scrum-мастер',
  'main.team.contributions-3': 'Дизайн, Jira, MUI, REST, Страница истории, Главная страница',
  'main.team.bio-3':
    'Окончила Балтийский федеральный университет имени Иммануила Канта, обучалась на факультете бизнес-информатики. Завершила курсы 3WC по специальности Frontend-разработчик. Разрабатывала функции на PHP для 1C CRM.',
  'main.rs-1':
    'RSSchool — это онлайн-образовательное сообщество, ориентированное на веб-разработку и программирование, предлагающее хорошо структурированные курсы как для начинающих, так и для опытных разработчиков.',
  'main.rs-2':
    'Известная своим высоким качеством образования, RSSchool предоставляет комплексную программу, охватывающую всё — от основ HTML, CSS и JavaScript до продвинутых технологий, таких как React и Node.js. Школа делает акцент на обучении через проекты, позволяя студентам создавать реальные проекты и развивать впечатляющие портфолио.',
  'main.rs-3':
    'RSSchool процветает благодаря активному сообществу волонтёров, включая опытных разработчиков и выпускников, которые наставляют и поддерживают новых учащихся.',

  'teamMember.role': 'Роль',
  'teamMember.contributions': 'Вклад',
  'teamMember.bio': 'Краткая биография',
  'teamMember.techStack': 'Стек технологий: ',

  'sign-in.loader': 'Входим...',
  'sign-in.title': 'Вход',
  'sign-in.email': 'Почта',
  'sign-in.password': 'Пароль',
  'sign-in.submit': 'Войти',

  'sign-up.loader': 'Регистрируем...',
  'sign-up.title': 'Регистрация',
  'sign-up.name': 'Имя',
  'sign-up.email': 'Почта',
  'sign-up.password.title': 'Пароль',
  'sign-up.password': 'Введите пароль',
  'sign-up.password.confirm': 'Подтвердите пароль',
  'sign-up.submit': 'Зарегистрироваться',

  'rest.title': 'REST клиент',
  'rest.endpoint': 'Эндпоинт URL',
  'rest.params': 'Параметры',
  'rest.headers': 'Заголовки',
  'rest.variables': 'Переменные',
  'rest.body.title': 'Тело: ',
  'rest.body': 'Тело',

  'restApi.errors.fetch': 'Не удалось получить ответ',
  'restApi.errors.endpoint': 'Эндпоинт не указан',
  'restApi.errors.body': 'Тело ответа содержит ошибки',
  'restApi.errors.unknown': 'Произошла неизвестная ошибка при выполнении запроса',
  'restApi.errors.status': 'Ошибка получения данных',

  method: 'Метод',

  'clientTable.hide': 'Скрыть',
  'clientTable.show': 'Показать',
  'clientTable.key': 'Ключ',
  'clientTable.value': 'Значение',
  'clientTable.row': '+ строка',
  'clientTable.delete': 'Удалить',

  'requestButton.send': 'Отправить Запрос',

  'response.title': 'Ответ',
  'response.status': 'Статус:',
  'response.body': 'Тело:',

  'graphql.errors.parse': 'Не удается разобрать тело запроса в URL',
  'graphql.schema.completed': 'Запрос выполнен, можете открыть схему',
  'graphql.request.completed': 'Запрос выполнен, проверьте поле ответа',
  'graphql.title': 'GraphQL клиент',
  'graphql.endpoint': 'Эндпоинт URL',
  'graphql.schema': 'Схема URL',
  'graphql.request': 'Запрос',
  'graphql.schema.request': 'Запрос схемы',
  'graphql.schema.hide': 'Скрыть схему',
  'graphql.schema.show': 'Показать схему',
  'graphql.params': 'Параметры',
  'graphql.query.title': 'Запрос данных: ',
  'graphql.query': 'Запрос данных',
  'graphql.headers': 'Заголовки',
  'graphql.variables.title': 'Переменные: ',
  'graphql.variables': 'Переменные',

  'schema.title': 'Документация по схеме:',

  'graphQlApi.response-errors.endpoint': 'Эндпоинт или запрос данных не указаны',
  'graphQlApi.response-errors.variables': 'Поле переменных не является валидным JSON',
  'graphQlApi.response-errors.data':
    'Неизвестная структура ответа: отсутствует поле "data". Проверьте эндпоинт и параметры запроса',
  'graphQlApi.response-errors.unknown': 'Произошла неизвестная ошибка при выполнении запроса',
  'graphQlApi.response-errors.status': 'Ошибка получения данных',

  'graphQlApi.schema-errors.endpoint': 'Эндпоинт схемы не указан',
  'graphQlApi.schema-errors.body': 'Тело ответа схемы содержит ошибки',
  'graphQlApi.schema-errors.data':
    'Неизвестная структура ответа: отсутствует поле "data" или "__schema". Проверьте эндпоинт и заголовки',
  'graphQlApi.schema-errors.fetch-status': 'Не удалось получить документацию схемы, код состояния: {status}',
  'graphQlApi.schema-errors.unknown': 'Произошла неизвестная ошибка при выполнении запроса документацию схемы',

  'history.title': 'История',
  'history.empty.title': 'Вы не выполняли никаких запросов.',
  'history.empty': 'Здесь пусто. Попробуйте: ',
  'history.links.rest': 'REST клиент',
  'history.links.graphql': 'GraphQL клиент',
  'history.subtitle': 'Запросы:',

  'error.title': 'Упс!',
  'error.text': 'Извините, произошла непредвиденная ошибка.',
  'error.desc': 'Сообщение об ошибке:',
  'error.recommendation': 'Пожалуйста, попробуйте обновить страницу.',
  'error.refresh': 'Обновить страницу',

  '404.message': 'Запрашиваемая вами страница не найдена.',
  '404.back': 'Предыдущая страница',
  '404.main': 'Главная страница',

  'auth.loading': 'Загружаем Firebase...',
  'auth.expired': 'Срок действия вашего токена истек, пожалуйста, войдите снова',
  'auth.redirect': 'Перенаправляем...',

  'firebase.sign-out.success': 'Вы вышли из системы. История запросов очищена!',
  'firebase.sign-in.success': 'Успешный вход с помощью {email}!',
  'firebase.sign-up.success': '{name} успешно зарегистрирован!',
  'firebase.errors.USER_DISABLED': 'Пользователь заблокирован',
  'firebase.errors.INVALID_LOGIN_CREDENTIALS': 'Неверные учетные данные',
  'firebase.errors.EMAIL_EXISTS': 'Почта уже используется',
  'firebase.errors.TOO_MANY_ATTEMPTS_TRY_LATER': 'Слишком много попыток входа',
  'firebase.errors.NETWORK_REQUEST_FAILED': 'Сетевой запрос не выполнен',

  'validation.email.required': 'Почта обязательна',
  'validation.email.format': 'Почта должна иметь допустимый формат',
  'validation.password.required': 'Пароль обязателен',
  'validation.password.number': 'Пароль должен содержать цифру',
  'validation.password.uppercase': 'Пароль должен содержать заглавную букву',
  'validation.password.lowercase': 'Пароль должен содержать строчную букву',
  'validation.password.special': 'Пароль должен содержать один из: -+/%*:#@\\$!?|^&',
  'validation.password.min': `Пароль должен быть не менее ${MIN_PASSWORD_LENGTH} символов`,
  'validation.name.required': 'Имя обязательно',
  'validation.name.capital': 'Имя должно начинаться с заглавной буквы',
  'validation.confirm.required': 'Пожалуйста, подтвердите свой пароль',
  'validation.confirm.match': 'Пароли не совпадают',
} as const;
