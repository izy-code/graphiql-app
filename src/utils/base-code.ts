export const encodeBase64 = (str: string): string => Buffer.from(str, 'utf-8').toString('base64');
export const decodeBase64 = (str: string): string => Buffer.from(str, 'base64').toString('utf-8');
