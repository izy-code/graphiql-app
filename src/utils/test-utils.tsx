import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { type PropsWithChildren, type ReactElement, useRef } from 'react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '@/store/store';
import { setupStore } from '@/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProvidersAndUser(
  ui: ReactElement,
  { preloadedState = {}, store = setupStore(preloadedState), ...renderOptions }: ExtendedRenderOptions = {},
): { store: AppStore; user: ReturnType<typeof userEvent.setup>; container: HTMLElement } {
  function Wrapper({ children }: PropsWithChildren): ReactElement {
    const storeRef = useRef<AppStore>();

    if (!storeRef.current) {
      storeRef.current = setupStore(preloadedState);
    }

    return <Provider store={storeRef.current}>{children}</Provider>;
  }

  return { store, user: userEvent.setup(), ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderWithUserSetup(ui: ReactElement): {
  user: ReturnType<typeof userEvent.setup>;
  container: HTMLElement;
} {
  return {
    user: userEvent.setup(),
    ...render(ui),
  };
}
