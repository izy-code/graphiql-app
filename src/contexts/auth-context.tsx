'use client';

import { type User } from 'firebase/auth';
import { useRouter } from 'next/navigation';
import type { ReactNode } from 'react';
import { createContext, useEffect, useState } from 'react';
import { toast } from 'react-toastify';

import { Loader } from '@/components/loader/Loader';
import { auth, logOut } from '@/firebase/firebase';
import { useScopedI18n } from '@/locales/client';

export interface UserImpl {
  stsTokenManager: {
    expirationTime: number;
  };
}

export type AuthContextType = User | null;

const TOKEN_CHECK_INTERVAL_MS = 60 * 1000;

export const AuthContext = createContext<AuthContextType>(null);

export function AuthProvider({ children }: { children: ReactNode }): ReactNode {
  const [user, setUser] = useState<User | null>(null);
  const [isLoading, setIsLoading] = useState(true);
  const router = useRouter();
  const translate = useScopedI18n('auth');

  useEffect(() => {
    const checkTokenExpiration = async (currentUser: User): Promise<void> => {
      const userImpl = currentUser.toJSON() as UserImpl;
      const { expirationTime } = userImpl.stsTokenManager;
      const currentTime = new Date().getTime();

      if (currentTime >= expirationTime) {
        await logOut();
        toast.info(translate('expired'));
        router.push('/');
      }
    };

    const unsubscribe = auth.onIdTokenChanged(async (changedUser) => {
      setUser(changedUser);
      setIsLoading(false);

      if (changedUser) {
        await checkTokenExpiration(changedUser);
      }
    });

    const intervalId = setInterval(async () => {
      if (auth.currentUser) {
        await checkTokenExpiration(auth.currentUser);
      }
    }, TOKEN_CHECK_INTERVAL_MS);

    return (): void => {
      unsubscribe();
      clearInterval(intervalId);
    };
  }, [router, translate]);

  return (
    <AuthContext.Provider value={user}>
      {isLoading ? <Loader loaderText={translate('loading')} /> : children}
    </AuthContext.Provider>
  );
}
