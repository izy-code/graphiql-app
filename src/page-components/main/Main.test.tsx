import { screen } from '@testing-library/react';
import React from 'react';

import MainPage from '@/app/[locale]/page';
import { AuthProvider } from '@/contexts/auth-context';
import en from '@/locales/en';
import ru from '@/locales/ru';
import { renderWithProvidersAndUser } from '@/utils/test-utils';

let text: string = en['main.welcome'];
const translate = vi.fn(() => text);

vi.mock('@/locales/client', async () => {
  const actual = await vi.importActual('@/locales/client');

  return {
    ...actual,
    useScopedI18n: vi.fn(() => translate),
  };
});

describe('Main page', () => {
  it('renders correctly', () => {
    const { container } = renderWithProvidersAndUser(
      <AuthProvider>
        <MainPage />
      </AuthProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Show welcome message with en locale', () => {
    renderWithProvidersAndUser(
      <AuthProvider>
        <MainPage />
      </AuthProvider>,
    );

    expect(screen.getByText(text)).toBeInTheDocument();
  });

  it('Show welcome message with ru locale', () => {
    text = ru['main.welcome'];

    renderWithProvidersAndUser(
      <AuthProvider>
        <MainPage />
      </AuthProvider>,
    );

    expect(screen.getByText(text)).toBeInTheDocument();
  });
});
