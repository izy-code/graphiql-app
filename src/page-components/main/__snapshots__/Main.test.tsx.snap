// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main page > renders correctly 1`] = `
<div>
  <div
    class="_page_d7ba51"
  >
    <div
      class="_bigSection_d7ba51"
    >
      <h1
        class="_welcomeTitle_d7ba51"
      >
        REST/GraphiQL client
      </h1>
      <p>
        Welcome, please sign in.
      </p>
      <div
        class="_additionalInfo_d7ba51"
      >
        <a
          href="/sign-in"
        >
          Sign in
        </a>
        <a
          href="/sign-up"
        >
          Sign up
        </a>
      </div>
    </div>
    <div
      class="_container_d7ba51"
    >
      <h2
        class="_sectionTitle_d7ba51"
      >
        Project information
      </h2>
      <div
        class="_introParagraph_d7ba51"
      >
        Developed as the final assignment for a React course, this project showcases the coordinated effort of three dedicated participants.
      </div>
      <div
        class="_introParagraph_d7ba51"
      >
        Completed within a span of four weeks, the application is a result of the team's hard work and collaboration.
      </div>
      <div
        class="_introParagraph_d7ba51"
      >
        The application supports both RESTful and GraphQL APIs, featuring Firebase authentication, request history tracking, and language switching. Users can configure, send requests, and view responses.
      </div>
      <div
        class="_blockInfo_f2db44 "
      >
        <a
          href="https://github.com/izy-code"
          target="_blank"
        >
          <div
            class="_imageTeam_f2db44 _imageIzy_f2db44"
          />
        </a>
        <div
          class="_textInfo_f2db44"
        >
          <a
            class="_titleBlock_f2db44"
            href="https://github.com/izy-code"
            target="_blank"
          >
            <h2>
              Ivan Zhirebny
               
            </h2>
          </a>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              GitHub: 
            </span>
            <a
              class="_link_f2db44"
              href="https://github.com/izy-code"
              target="_blank"
            >
              izy-code
            </a>
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Role
              : 
            </span>
            Frontend-developer, Team leader
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Contributions
              : 
            </span>
            Dev environment setup, Authentication, Firebase
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Short bio
              : 
            </span>
             Graduated from the Bauman Moscow State Technical University, completed the Frontend Developer program at the HTML Academy
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Tech stack: 
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              CSS
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Sass
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              HTML
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              JavaScript
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Java
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              TypeScript
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Webpack
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Vite
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Jest
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              React
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Next
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Redux
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Remix
            </span>
          </div>
        </div>
      </div>
      <div
        class="_blockInfo_f2db44 _rotate_f2db44"
      >
        <a
          href="https://github.com/VadimKol"
          target="_blank"
        >
          <div
            class="_imageTeam_f2db44 _imageVadim_f2db44"
          />
        </a>
        <div
          class="_textInfo_f2db44"
        >
          <a
            class="_titleBlock_f2db44"
            href="https://github.com/VadimKol"
            target="_blank"
          >
            <h2>
              Vadim Kolymbet
               
            </h2>
          </a>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              GitHub: 
            </span>
            <a
              class="_link_f2db44"
              href="https://github.com/VadimKol"
              target="_blank"
            >
              vadimkol
            </a>
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Role
              : 
            </span>
            Frontend-developer
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Contributions
              : 
            </span>
            Routing, Internationalization
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Short bio
              : 
            </span>
            Graduated from the Saratov State University, studied at the Faculty of Computer Science and Information Technologies and in the Department of Computer Security and Cryptography Theory. Graduated from layout courses at Frontendblok. Worked as a T-SQL developer
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Tech stack: 
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              CSS
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Sass
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              SQL
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              HTML
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              JavaScript
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              TypeScript
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Webpack
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Vite
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Jest
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              React
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Next
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Redux
            </span>
          </div>
        </div>
      </div>
      <div
        class="_blockInfo_f2db44 "
      >
        <a
          href="https://github.com/BodnarAlex"
          target="_blank"
        >
          <div
            class="_imageTeam_f2db44 _imageAlex_f2db44"
          />
        </a>
        <div
          class="_textInfo_f2db44"
        >
          <a
            class="_titleBlock_f2db44"
            href="https://github.com/BodnarAlex"
            target="_blank"
          >
            <h2>
              Aleksandra Bodnar
               
            </h2>
          </a>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              GitHub: 
            </span>
            <a
              class="_link_f2db44"
              href="https://github.com/BodnarAlex"
              target="_blank"
            >
              bodnaralex
            </a>
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Role
              : 
            </span>
            Frontend-developer, Scrum-master
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Contributions
              : 
            </span>
            Jira, Design, MUI
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Short bio
              : 
            </span>
            Graduated from the Immanuel Kant Baltic Federal University, studied at the Faculty of Business Informatics. Graduated from 3WC courses with a degree in Frontend-development. Built PHP-driven features for 1C CRM
          </div>
          <div>
            <span
              class="_infoTitle_f2db44"
            >
              Tech stack: 
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              PHP
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              C++
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Python
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Wordpress
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              CSS
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Sass
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              HTML
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              JavaScript
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              TypeScript
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Webpack
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Vite
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              React
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Next
            </span>
            <span
              class="_softStackItem_f2db44"
            >
              Redux
            </span>
          </div>
        </div>
      </div>
      <div
        class="_blockInfo_f2db44 _rotate_f2db44"
      >
        <a
          href="https://rs.school"
          target="_blank"
        >
          <div
            class="_imageTeam_f2db44 _imageRss_f2db44"
          />
        </a>
        <div
          class="_textInfo_f2db44"
        >
          <a
            class="_titleBlock_f2db44"
            href="https://rs.school"
            target="_blank"
          >
            <h2>
              Rolling Scopes School
               
            </h2>
          </a>
          <div>
            RSSchool is an online educational community focused on web development and programming, offering well-structured courses for both beginners and experienced developers. 
          </div>
          <div>
            Renowned for its high-quality education, RSSchool provides a comprehensive curriculum covering everything from basic HTML, CSS, and JavaScript to advanced technologies like React and Node.js. The school emphasizes project-based learning, enabling students to build real-world projects and develop impressive portfolios.
          </div>
          <div>
            RSSchool thrives on a vibrant community of volunteers, including seasoned developers and alumni, who mentor and support new learners.
          </div>
        </div>
      </div>
    </div>
  </div>
  ;
</div>
`;
